application/src/main/java/org/thingsboard/server/service/install/update/DefaultDataUpdateService.java

```diff
@@ -564,6 +572,24 @@ private void updateOAuth2Params() {
         log.warn("CAUTION: Update of Oauth2 parameters from 3.2.2 to 3.3.0 available only in ThingsBoard versions 3.3.0/3.3.1");
     }
 
+    private final PaginatedUpdater<String, Tenant> tenantCustomersTitleUpdater =
+            new PaginatedUpdater<>() {
+
+                @Override
+                protected String getName() {
+                    return "Customer title updater";
+                }
+
+                @Override
+                protected PageData<Tenant> findEntities(String id, PageLink pageLink) {
+                    return tenantService.findTenants(pageLink);
```

Comment:
This is wrong. Why do you need to iterate over tenants? Just fetch ALL customers with duplicated names (grouped by tenant) and iterate over this list. If we have 50K tenants in the system, we will need to run 50K separate queries (1 per tenant). The goal is to run only one query.

Reply:
Thanks for review PR, 
I add commit with improve for this case